ports/nspr/portfile.cmake

```
@@ -0,0 +1,88 @@
+set(NSPR_VERSION "4.33")
+
+vcpkg_download_distfile(ARCHIVE
+    URLS "https://releases.mozilla.org/pub/nspr/releases/v${NSPR_VERSION}/src/nspr-${NSPR_VERSION}.tar.gz"
+    FILENAME "nspr-${NSPR_VERSION}.tar.gz"
+    SHA512 8064f826c977f1302a341ca7a7aaf7977b5d10102062c030b1d42b856638e3408ab262447e8c7cfd5a98879b9b1043d17ceae66fbb1e5ed86d6bc3531f26667e
+)
+
+vcpkg_extract_source_archive_ex(
+    OUT_SOURCE_PATH SOURCE_PATH
+    ARCHIVE ${ARCHIVE}
+    REF "${NSPR_VERSION}"
+)
+
+set(MOZBUILD_ROOT "${CURRENT_HOST_INSTALLED_DIR}/tools/mozbuild")
+
+set(MOZBUILD_BINDIR "${MOZBUILD_ROOT}/bin")
+vcpkg_add_to_path("${MOZBUILD_BINDIR}")
+
+set(MOZBUILD_MSYS_ROOT "${MOZBUILD_ROOT}/msys")
+vcpkg_add_to_path(PREPEND "${MOZBUILD_MSYS_ROOT}")
+
+set(OPTIONS "")
+if (VCPKG_CRT_LINKAGE STREQUAL "dynamic")
+    list(APPEND OPTIONS "--disable-static-rtl")
+else()
+    list(APPEND OPTIONS "--enable-static-rtl")
+endif()
+
+list(APPEND OPTIONS "--enable-win32-target=win95")
+
+if (VCPKG_TARGET_ARCHITECTURE STREQUAL "x64")
+    list(APPEND OPTIONS "--enable-64bit")
+elseif(VCPKG_TARGET_ARCHITECTURE STREQUAL "x86")
+    list(APPEND OPTIONS "--disable-64bit")
+endif()

```

```suggestion
else()
    message(FATAL_ERROR "Unrecognized VCPKG_TARGET_ARCHITECTURE.")
endif()

```

I see you don't care about arm et-al in your comments but it's still good to put defenses in here in case someone in the future tries changing the supports and doesn't realize they need to resolve that problem.